src/passes/TypeMerging.cpp

```diff
@@ -57,13 +57,21 @@ using ReferredTypes = SmallUnorderedSet<HeapType, 5>;
 struct CastFinder : public PostWalker<CastFinder> {
   ReferredTypes referredTypes;
 
+  bool trapsNeverHappen;
```

Comment:
Maybe add a comment as to why it is valid to ignore certain things when TNH, like in the PR description?